drivers/sdcard/sdcard.py

```diff
@@ -228,13 +228,15 @@ def readblocks(self, block_num, buf):
         assert nblocks and not len(buf) % 512, 'Buffer length is invalid'
         if nblocks == 1:
             # CMD17: set read address for single block
-            if self.cmd(17, block_num * self.cdv, 0) != 0:
+            if self.cmd(17, block_num * self.cdv, 0, release=False) != 0:
```

Comment:
If `self.cmd` returns non-zero but not an error (not -1) then the CS won't be released.  Or maybe it's guaranteed that CMD17 returns either 0 or fails?

Reply:
I somehow thought that raising an exception will handle this. Added release of cs if result is != 0.